### ISSUE TITLE
KerasClassifier problem when a label is not present in a particular split

### ISSUE THREAD

**Issue by dmitriydligach** [keras-team/keras#11850]
This is an issue with Keras scikit-learn wrapper which I am using with sklearn.model_selection.RandomizedSearchCV(...). 

KerasClassifier.fit(self, x, y, ...) assumes that all labels that a model can predict exist in the training data (i.e. in 'y'):

// line 203
self.classes_ = np.unique(y)

However, in my case the model has three outputs but a particular split sometimes contains only two labels because the third label is rare. Because the model sometimes still predicts that label that's not in self.classes_, I get an error causes by this line in KerasClassifier.predict(...):

// line 234
return self.classes_[classes]

I verified that replacing the line:
self.classes_ = np.unique(y)

with this line:
self.classes_ = np.array([0, 1, 2])

fixes the problem.

**Comment by dmitriydligach** [comment]
I verified that replacing the line: 
self.classes_ = np.unique(y)

with this line:
self.classes_ = np.array([0, 1, 2])

fixes the problem.

**Comment by liebscher** [comment]
I would look into a [Stratified KFold ](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html) routine instead of the typical KFold CV. With RandomizedSearch you can specify your own CV routine.

**Comment by dmitriydligach** [comment]
@liebscher Thanks. This may or may not help. In some cases there's only one example of the third class, so one of the folds will unavoidably contain only two classes. It seems like this is a bug that should be addressed. Maybe KerasClassifier should have a way to specify the number of classes in the data or something like that.

**Comment by liebscher** [comment]
@dmitriydligach you're right, when there's only one sample under the label even Stratified KFold will throw an exception. If there are cases where there's only one sample in the data set with a certain label, you may be very hard pressed to train a model to learn it. Hopefully someone else sees this issue and has a more optimistic solution!
